app/ActiveCampaign.php

```diff
@@ -56,17 +66,63 @@ public function get($dealId)
         $deal = $data['deal'];
 
         // Fetch custom fields.
-        $data = $this->call('GET', 'deals/' . $dealId . '/dealCustomFieldData');
-        if (!isset($data['dealCustomFieldData'])) {
-            throw new RuntimeException('Could not get deal custom fields on ' . $dealId);
-        }
-        foreach ($data['dealCustomFieldData'] as $customField) {
-            if (!isset($customField['customFieldId']) || !isset($customField['fieldValue'])) {
+        $data = $this->getCustomFields($dealId);
+        foreach ($data as $customField) {
+            if (!isset($customField['customFieldId']) ||
+                !array_key_exists('fieldValue', $customField)) {
+                print_r($customField);
                 throw new RuntimeException('Malformed custom field response on deal ' . $dealId);
             }
             $deal['custom_field_' . $customField['customFieldId']] = $customField['fieldValue'];
         }
 
         return $deal;
     }
+
+    public function updateCustomField($dealId, $fieldName, $value)
+    {
+        if (!preg_match('/^custom_field_(\d+)$/', $fieldName, $matches)) {
+            throw new RuntimeException('Bad custom field id: ' . $id);
+        }
+        $fieldId = $matches[1];
+
+        // Find field id for the field instance on the deal.
+        $fieldInstanceId = null;
+        $data = $this->getCustomFields($dealId);
```

Comment:
samme her. kan data være tom? Så skal det nok wrappes i et tjek inden for each

Reply:
Får `getCustomFields` til at sørge for at det er en array. Om den så skal insistere på at APIet skal returnere en tom array hvis der ikke er nogen fields, eller om den bare skal checke om den har fået noget `empty` er spørgsmålet.